hashicorp/terraform-provider-tfe
# support for organization-level default execution mode and agent pool

## Description

Adds support for organization-level default execution modes and agent pools 🎉 

_Remember to:_


## Testing plan

1. Verify that workspaces can be created/updated to utilize the organization's default execution mode while the organization default execution mode is set to `agent`, then again with the default set to `remote`. Flip back and forth between the execution mode being set at the workspace-level and the organization-level
1. Verify that the workspace's `operations` setting still works as expected, and that workspaces created with the `operations` setting can be migrated to use the organization default execution mode. Do this by creating a workspace with `operations` set to some value. Then remove the operations setting and apply the configuration.
1. Using an older version of TFE, verify that the behavior of the provider matches the old behavior (`remote` is always the default execution mode).

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [go-tfe PR](https://github.com/hashicorp/go-tfe/pull/762)
- [RFC](https://docs.google.com/document/d/19Y-OHqB76fmidbTaZq0_FEEIfE4YgBRC9gUW5WUvtM0)
